services/src/main/java/org/fao/geonet/api/records/InspireValidationApi.java

```diff
@@ -0,0 +1,237 @@
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+import javax.xml.transform.TransformerConfigurationException;
+import javax.xml.transform.TransformerException;
+import javax.xml.transform.TransformerFactoryConfigurationError;
+
+import org.apache.http.HttpStatus;
+import org.fao.geonet.ApplicationContextHolder;
+import org.fao.geonet.api.API;
+import org.fao.geonet.api.ApiParams;
+import org.fao.geonet.api.ApiUtils;
+import org.fao.geonet.api.exception.ResourceNotFoundException;
+import org.fao.geonet.api.records.editing.InspireValidatorUtils;
+import org.fao.geonet.constants.Geonet;
+import org.fao.geonet.domain.Metadata;
+import org.fao.geonet.kernel.DataManager;
+import org.fao.geonet.kernel.SchemaManager;
+import org.fao.geonet.kernel.setting.SettingManager;
+import org.fao.geonet.kernel.setting.Settings;
+import org.fao.geonet.utils.Log;
+import org.jdom.Attribute;
+import org.jdom.Document;
+import org.jdom.Element;
+import org.jdom.output.XMLOutputter;
+import org.springframework.context.ApplicationContext;
+import org.springframework.http.MediaType;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+import javassist.NotFoundException;
+import jeeves.server.context.ServiceContext;
+import jeeves.services.ReadWriteController;
+import springfox.documentation.annotations.ApiIgnore;
+
+@RequestMapping(value = {
+        "/api/inspire",
+        "/api/" + API.VERSION_0_1 +
+        "/inspire"
+})
+@Api(value = API_CLASS_RECORD_TAG,
+tags = API_CLASS_RECORD_TAG)
+@Controller("inspire")
+@PreAuthorize("hasRole('Editor')")
+@ReadWriteController
+public class InspireValidationApi {
+
+    @ApiOperation(
+            value = "Submit a record to the INSPIRE service for validation.",
+            notes = "User MUST be able to edit the record to validate it. "
+                    + "An INSPIRE endpoint must be configured in Settings. "
+                    + "This activates an asyncronous process, this method does not return any report. "
+                    + "This method returns an id to be used to get the report.",
+                    nickname = "submitValidate")
+    @RequestMapping(value = "/{metadataUuid}/validate/submit",
```

Comment:
Would it be better to have the API like 
```
PUT /srv/api/0.1/records/{metadataUuid}/validate/internal (current one XSD + schematron)
PUT /srv/api/0.1/records/{metadataUuid}/validate/inspire
```
?

So we can add new validator in the long term.